caringcaribou/modules/xcp.py

```diff
@@ -273,14 +279,20 @@ def connect_callback_handler(msg):
             def callback_handler(msg):
                 global command_reply
                 if msg.arbitration_id == rcv_arb_id:
-                    print("{0:<23} {1}".format(cmd_desc, msg.data[0] != 0xfe))
+                    if msg.data[0] == 0xFF:
+                        print("{0:<23} AVAILABLE".format(cmd_desc))
+                    elif (msg.data[0] == 0xFE) and (msg.data[1] == 0x20):
+                        print("{0:<23} NOT AVAILABLE".format(cmd_desc))
+                    else:
+                        print("{0:<23} {1} ({2})".format(cmd_desc,XCP_ERROR_CODES[msg.data[1]][0],
+                                                         hex(msg.data[1])))
```

Comment:
Same thing here with message length as above - length check needed before accessing index 1. This also means there will be a need for a final `else` case or other handling, when the conditions on this one are not met (i.e. the corner case when `msg.data[0]` is not `0xFF` and `msg.data` is only 1 byte long).